### ISSUE TITLE
Include quality bits in digitized clusters sent to the Correlator

### ISSUE THREAD

**Issue by skkwan** [skkwan/phase2-l1Calo-analyzer#14]
> Also, I see that the egamma digitized clusters sent to the Correlator:
> https://github.com/cms-l1t-offline/cmssw/blob/l1t-integration-CMSSW_13_2_0_pre3/DataFormats/L1TCalorimeterPhase2/interface/DigitizedClusterCorrelator.h
> 
> Currently, do not encode any quality bit. This is a showstopper to start using them in the emulator since any flag used in the GT decision must be propagated.
> (e.g. looseTkEleId)

This was the previous correlator digitized cluster format:
![image](https://github.com/skkwan/phase2-l1Calo-analyzer/assets/20006539/b234f37c-0c7e-4b55-8661-3624c14a6d1f)

The correlator digitized cluster format has been updated.

![image](https://github.com/skkwan/phase2-l1Calo-analyzer/assets/20006539/9ec6cf8c-4bff-498a-84c9-a3e9ff9b1bbf)

![image](https://github.com/skkwan/phase2-l1Calo-analyzer/assets/20006539/963e38b1-a701-4a41-9e88-82593dc93954)

**Comment by skkwan** [comment]
- [x] `pT` bits are the same
- [x] `eta` bits are now 8 bits instead of 6 (19-12, instead of 17-12) <--- and are now the full eta information.
   - LSB is 2.8/170
- [x] `phi` bits are 7 bits instead of 5 (26-20, instead of 22-18) <--- and are now the full phi information
   - LSB is 2pi/360
- [x] `H/E` value is still 4 bits, and is now 30-27, instead of 31-28 
- [x] `H/E flag` is new, 2 bits, and is 32-31 
- [x] `Isolation value` is still 3 bits, and is now 34-32 instead of 35-33
- [x] `Isolation flag` is new, 2 bits, and is 37-36
- [x] `FB` is still 6 bits, and is now 43-38 instead of 40-35
- [x] `Timing` is still 5 bits, and is now 48-44 instead of 45-41
- [x] `Shape flag` is now 2 bits instead of 1 bit, and is bits 50-49 instead of bit 46
- [x] `Brems` is now 2 bits instead of 1 bit, and is bits 52-51 instead of bit 47
- [x] `Spare` bits are 63-53, instead of 63-48

**Comment by skkwan** [comment]
Checks for one event for `DigitizedClusterCorrelator`:

- [x] pT of leading clusters match GCT emulator outputs
- [x] Shower shape and iso flags match with GCT emulator outputs
- [x] Real eta of leading clusters match GCT emulator outputs
- [x] Real phi of leading clusters match GCT emulator outputs

One event: `event 58098` in  `root://cms-xrd-global.cern.ch///store/mc/Phase2Spring23DIGIRECOMiniAOD/DoubleElectron_FlatPt-1To100-gun/GEN-SIM-DIGI-RAW-MINIAOD/PU200_Trk1GeV_131X_mcRun4_realistic_v5-v1/30000/0083e06c-959e-4ea5-9190-fd73667fcc00.root` 

These are the two gen electrons:
```bash
>>> Corrected gen electron pt/eta/phi: (20.7481, -1.31399, -0.228769) to 20.7481, -1.31816, -0.191478 with ecal_pos eta/phi -1.31816, -0.191478
>>> Corrected gen electron pt/eta/phi: (20.7481, 1.31399, 2.91282) to 20.7481, 1.31021, 2.87553 with ecal_pos eta/phi 1.31021, 2.87553
```

Corresponding to these deltaR matched GCT clusters:
```
--> Matched GCT cluster 16 eta: -1.31823 phi: -0.200709 with genElectron 20.7481 eta: -1.31816 phi: -0.191478.  iso: 0.09375 et2x5: 9.5 et5x5: 9.5 is_ss: 1 is_looseTkss: 1 is_iso: 1 is_looseTkiso: 1
--> Matched GCT cluster 11 eta: 1.30077 phi: 2.90598 with genElectron 20.7481 eta: 1.31021 phi: 2.87553.  iso: 0.318182 et2x5: 7 et5x5: 7 is_ss: 1 is_looseTkss: 1 is_iso: 1 is_looseTkiso: 1
```

Which correspond to these two digitized clusters:
```bash
>> > Correlator digitized cluster found with pT in GeV: 16 crystal eta 9 crystal phi 48 iso flags 3 shower shape flags 3 passes iso, looseTkiso, passes ss, looseTkss 1 1 1 1, realEta: -1.32696, real Phi: -0.209439
>> > Correlator digitized cluster found with pT in GeV: 11 crystal eta 159 crystal phi 106 iso flags 3 shower shape flags 3 passes iso, looseTkiso, passes ss, looseTkss 1 1 1 1, realEta: 1.29204, real Phi: 2.89725
```

Checking some non-leading clusters in the event: this GCT cluster
```bash
GCT Cluster found: pT 9.5, eta 0.14841, phi -1.00356,  with flags: is_ss 0,is_iso 1, is_looseTkss 1, is_looseTkiso 1
```

corresponds to this digitized cluster, and their flags agree:
```bash
>> > Correlator digitized cluster found with pT in GeV: 9.5 crystal eta 93 crystal phi 2 iso flags 3 shower shape flags 2 passes iso, looseTkiso, passes ss, looseTkss 1 1 0 1, realEta: 0.13968, real Phi: -1.01229
```
